grammar.y

```diff
@@ -972,7 +986,35 @@ create_table_stmt:
     if len($5) > MaxAllowedColumns {
       yylex.(*Lexer).AddError(&ErrTooManyColumns{ColumnCount: len($5), MaxAllowed: MaxAllowedColumns})
     }
-    $$ = &CreateTable{Table: $3, Columns: $5, Constraints: $6}
+
+    // We have to replace a primary key table constraint with an equivalent column constraint primary key,
+    // so we can add the autoincrement flag, as part of the rules of the Tableland Protocol.
+    // 
+    // That happens because a primary key table constraint that references a single INTEGER column
+    // would be an alias to rowid. For cases where a column becomes an alias to rowid we want to force the AUTOINCREMENT.
+    // 
+    // The exception to the above rule is when a table constraint primary key has order DESC. In that case, we replace with an 
+    // equivalent column constrain without forcing AUTOINCREMENT.
+    var tableConstraintPK *TableConstraintPrimaryKey
+    var index int
+    for i, tableConstraint := range $6 {
+      if tcpk, ok := tableConstraint.(*TableConstraintPrimaryKey); ok && len(tcpk.Columns) == 1 {
+        tableConstraintPK = tcpk
+        index = i
+      }
+    }
+
+    for _, columnDef := range $5 {
```

Comment:
I think we can move all this `for` section inside the above `if`, since all this work only makes sense if `tableConstraintPK` isn't `nil`. 

So, I think if we move this section inside L1003-L1004 we can:
- Remove that `tableConstraintPK != nil` condition in L1010.
- Avoid iterating `range $5` if there're no table constraints.
- We can remove the `index` variable and use `i` directly.

Reply:
nice. good suggestion!